### ISSUE TITLE
Implement `cm validate tier` command

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#96]
## Description

Implement the `cm validate tier` command that verifies a project's check.toml extends from rulesets appropriate for its tier (production, internal, prototype).

## Parent Epic

Part of #94

## CLI Interface

```bash
cm validate tier         # Validate tier-ruleset alignment
cm validate tier --json  # JSON output
```

## Validation Rules

1. If `tier: production` → rulesets must include a `*-production` ruleset
2. If `tier: internal` → rulesets must include a `*-internal` ruleset
3. If `tier: prototype` → rulesets must include a `*-prototype` ruleset
4. Overrides are allowed, but base ruleset must match tier
5. If no tier specified, default to `internal`

## Tasks

- [ ] Add `tier` subcommand to `cm validate`
- [ ] Read tier from `repo-metadata.yaml`
- [ ] Read rulesets from check.toml `[extends]` section
- [ ] Implement tier-ruleset matching logic
- [ ] Add human-readable output with success/failure messages
- [ ] Add `--json` output format
- [ ] Export `validateTierRuleset()` programmatic API

## Definition of Done

- [ ] `cm validate tier` correctly validates alignment
- [ ] Mismatch errors provide actionable guidance
- [ ] Tests passing
